contrib/hived-config-patcher/src/patcher.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import sys
+import yaml
+import logging
+import argparse
+from kubernetes import client, config, watch
+
+
+class HiveDConfigPatcher(object):
+    def __init__(self, min_nodes=1, max_nodes=2**31-1, hived_config_file=None):
+        logging.basicConfig(level=logging.INFO, format="%(asctime)s [%(levelname)s] %(message)s")
+        self.min_nodes = min_nodes
+        self.max_nodes = max_nodes
+        self.hived_config_file = hived_config_file
+        self.__init_hived_config()
+        self.__init_kube_client()
+        self.__nodes = []
+        self.__fake_nodes = ["fake{}".format(self.base36(i)) for i in range(max_nodes)]
+
+    def __init_hived_config(self):
+        if self.hived_config_file is None:
+            logging.error("Cannot find hived config file, exit.")
+            sys.exit(1)
+        try:
+            with open(self.hived_config_file, "r") as f:
+                self.__hived_config = yaml.load(f)
+        except Exception:
+            logging.error("Cannot load hived config, exit.")
+            sys.exit(1)
+        if ("physicalCluster" not in self.__hived_config or
+                "physicalCells" not in self.__hived_config["physicalCluster"] or
+                len(self.__hived_config["physicalCluster"]["physicalCells"]) == 0):
+            logging.error("Invalid hived config, exit.")
+            sys.exit(1)
+
+    def __init_kube_client(self):
+        try:
+            config.load_kube_config("../config.log")
```

Comment:
What's this? You can use in cluster config

Reply:
updated, it was for local debug